crates/stages/src/stages/execution.rs

```diff
@@ -418,17 +424,33 @@ impl<EF: ExecutorFactory, DB: Database> Stage<DB> for ExecutionStage<EF> {
         let mut stage_checkpoint = input.checkpoint.execution_stage_checkpoint();
 
         // Unwind all receipts for transactions in the block range
-        let mut cursor = tx.cursor_write::<tables::Receipts>()?;
-        let mut reverse_walker = cursor.walk_back(None)?;
-
-        while let Some(Ok((tx_number, receipt))) = reverse_walker.next() {
-            if tx_number < first_tx_num {
-                break
+        match maybe_snapshotter(&self.prune_modes, provider, *range.start())? {
+            Some(_) => {
+                // `maybe_snapshotter` already prunes receipts if it detects that there are more
+                // elements in the static files than expected.
+                if let Some(stage_checkpoint) = stage_checkpoint.as_mut() {
+                    for block_number in range {
```

Comment:
this is to "unwind" the stage progress?

Reply:
added more context, lmk if it makes sense